From 6c963fb60da75e53214579507d7717acadb08c0f Mon Sep 17 00:00:00 2001
From: BtbN <btbn@btbn.de>
Date: Mon, 7 Sep 2020 20:33:23 +0200
Subject: [PATCH 3/4] Define appropiate minimum Windows-Version

---
 loader/vk_loader_platform.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/loader/vk_loader_platform.h b/loader/vk_loader_platform.h
index 2e138aeaa..d956e803a 100644
--- a/loader/vk_loader_platform.h
+++ b/loader/vk_loader_platform.h
@@ -30,6 +30,11 @@
 #endif
 
 #if defined(_WIN32)
+
+#define WINVER 0x0601
+#define _WIN32_WINNT 0x0601
+#define STRSAFE_NO_DEPRECATE 1
+
 // WinSock2.h must be included *BEFORE* windows.h
 #include <winsock2.h>
 #endif  // _WIN32
-- 
2.25.1

